lib/portage/dbapi/bintree.py

```diff
@@ -1400,6 +1554,10 @@ def _pkgindex_entry(self, cpv):
 		"""
 
 		pkg_path = self.getname(cpv)
+		if hasattr(cpv, "_metadata"):
+			binpkg_format = cpv._metadata.get('BINPKG_FORMAT', "xpak")
+		else:
+			binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
```

Comment:
The callers of _pkgindex_entry always read the metadata earlier with _read_metadata, so we should raise an error here if cpv._metadata is not available for some reason. We don't want to use `self.settings["BINPKG_FORMAT"]` here because that value can be different from the format of the current package.